packages/purpose-process/test/testGetPurposes.ts

```diff
@@ -231,19 +231,47 @@ export const testGetPurposes = (): ReturnType<typeof describe> =>
           eservicesIds: [],
           consumersIds: [],
           producersIds: [],
-          states: [purposeVersionState.rejected, purposeVersionState.archived],
+          states: [purposeVersionState.rejected, purposeVersionState.active],
           excludeDraft: undefined,
         },
         { offset: 0, limit: 50 },
         genericLogger
       );
-      expect(result.totalCount).toBe(3);
+      expect(result.totalCount).toBe(4);
       expect(result.results).toEqual([
+        mockPurpose3,
         mockPurpose4,
         mockPurpose5,
         mockPurpose6,
       ]);
     });
+    it("should get the purposes with only archived versions (and exclude the ones with both archived and non-archived versions)", async () => {
+      const mockArchivedVersion: PurposeVersion = {
+        ...getMockPurposeVersion(),
+        state: purposeVersionState.archived,
+      };
+      const mockArchivedPurpose: Purpose = {
+        ...getMockPurpose(),
+        title: "archived purpose",
+        eserviceId: mockEService1ByTenant1.id,
+        versions: [mockArchivedVersion],
+      };
+      await addOnePurpose(mockArchivedPurpose, postgresDB, purposes);
```

Comment:
may I ask to add a purpose that has one archived version and one non archived version to confirm the correct filter behaviour?

Reply:
Updated